crucible/src/Lang/Crucible/Backend/Online.hs

```diff
@@ -458,7 +457,7 @@ withSolverProcess ::
   IO a {- ^ Default value to return if online features are disabled -} ->
   (SolverProcess scope solver -> IO a) ->
   IO a
-withSolverProcess = withSolverProcess' (\sym -> readIORef (B.sbStateManager sym))
+withSolverProcess = withSolverProcess' (\sym -> pure (B.sbUserState sym))
```

Comment:
The `withSolverProcess'` is also used in crucible-saw (src/Lang/Crucible/Backend/SAWCore.hs, line ~1178), which didn't get updated yet (unless I missed it in this PR).

Also, since `withSolverProcess'` is only called from here and in crucible-saw, where it doesn't need to be in the IO monad either, should the argument just be changed to be pure (and so for example, here the value is just `B.sbUserState` instead of `(\sym -> ...)`?

Reply:
`crucible-saw` is now obsolete and is being allowed to bit-rot.  It's been removed from the `cabal.project`.

The `withSolverProcess'` stuff was some awkward pipe-fitting that I think is not needed anymore.  Maybe I should just get rid of it.